packages/backend/src/services/CoderService/CoderService.ts

```diff
@@ -0,0 +1,180 @@
+import { subject } from '@casl/ability';
+import {
+    CreateSavedChart,
+    NotFoundError,
+    SavedChartDAO,
+    SessionUser,
+    SpaceSummary,
+    UpdatedByUser,
+} from '@lightdash/common';
+import { ChartAsCode, currentVersion } from '@lightdash/common/src/types/coder';
+import { LightdashAnalytics } from '../../analytics/LightdashAnalytics';
+import { LightdashConfig } from '../../config/parseConfig';
+import { DashboardModel } from '../../models/DashboardModel/DashboardModel';
+import { ProjectModel } from '../../models/ProjectModel/ProjectModel';
+import { SavedChartModel } from '../../models/SavedChartModel';
+import { SpaceModel } from '../../models/SpaceModel';
+import { SchedulerClient } from '../../scheduler/SchedulerClient';
+import { BaseService } from '../BaseService';
+
+type CoderServiceArguments = {
+    lightdashConfig: LightdashConfig;
+    analytics: LightdashAnalytics;
+    projectModel: ProjectModel;
+    savedChartModel: SavedChartModel;
+    dashboardModel: DashboardModel;
+    spaceModel: SpaceModel;
+    schedulerClient: SchedulerClient;
+};
+
+export class CoderService extends BaseService {
+    lightdashConfig: LightdashConfig;
+
+    analytics: LightdashAnalytics;
+
+    projectModel: ProjectModel;
+
+    savedChartModel: SavedChartModel;
+
+    dashboardModel: DashboardModel;
+
+    spaceModel: SpaceModel;
+
+    schedulerClient: SchedulerClient;
+
+    constructor({
+        lightdashConfig,
+        analytics,
+        projectModel,
+        savedChartModel,
+        dashboardModel,
+        spaceModel,
+        schedulerClient,
+    }: CoderServiceArguments) {
+        super();
+        this.lightdashConfig = lightdashConfig;
+        this.analytics = analytics;
+        this.projectModel = projectModel;
+        this.savedChartModel = savedChartModel;
+        this.dashboardModel = dashboardModel;
+        this.spaceModel = spaceModel;
+        this.schedulerClient = schedulerClient;
+    }
+
+    private static transformChart(
```

Comment:
converts a complex SavedChartDAO into a `ChartAsCode` simple type, by removing some extra details, uuids , and adding some chartAsCode specific stuff like version or spaceSlug.